### ISSUE TITLE
Load times doubled, very long time @Adding itemStack ingredients. -

### ISSUE THREAD

**Issue by iTraumatik** [mezz/JustEnoughItems#544]
@mezz     Before your request to start pulling out 1 mod @ a time, I made new instance in MultiMC clean. Forge 2099, JEI 3.13.0.334,  8gb ram, and started putting mods in 1 by one.  I got up to  38 and then put ActuallyAdditions-1.10.2-r67 in and thats when it counts almost always up by 1 with a couple small skips up (but this is when it took way longer).   I went back to my normal instance and took out AAdditions and it still counts up very slow.
  NO Optifine in either instance.
This is what the log shows when Adding ItemStack ingredients is counting up
"[19:08:12] [Client thread/INFO]: Built    recipe registry in 2959 ms
[19:08:12] [Client thread/INFO]: Building runtime...
[19:08:12] [Client thread/INFO]: Built    runtime in 47 ms
[19:08:12] [Client thread/INFO]: Building item filter..."
In windows task manager java shows using 4149.2mb then after a minute it goes to 4152.4mb. Title screen 6175mb java used. In game F3 up to 4100mb 7485mb in task manager.  No lag.
19,619 total ingredients.

**Comment by mezz** [comment]
Thanks for the report!
Please paste your full client log to https://gist.github.com and link it here.

**Comment by iTraumatik** [comment]
@mezz   https://gist.github.com/iTraumatik/8ccd3ddde3a09d96fefced7274e7c889

**Comment by mezz** [comment]
Please try keeping just JEI and Actually Additions, then adding mods until it has the slowdown problem.
I think that another mod in your pack has the same problem as Optifine.

**Comment by iTraumatik** [comment]
@mezz   OK finally , added Endercore and Ender IO and right from the start its slow counting up in almost 1 increments.  Took out EnderIO and it still does it. So anybody with long loads EnderCore. Im using the latest off of curseforge. Both / all marked as beta , but arent all mods beta :)    

EDIT.. ok took out EnderCore and EnderIO from main instance, and its still doing it. So far i took out AA and now EnderIO and Core in main instance. And no Optifine in either.

**Comment by mezz** [comment]
Thanks for narrowing it down!
There is still some more to figure out though, my game still loads very quickly with those mods. 

I am using only:
Forge 12.18.2.2114
Actually Additions r67
EnderCore 0.4.1.56-beta
JEI 3.13.0.334 beta

If those are slow for you, can you upload your client log again, with just those mods?

**Comment by iTraumatik** [comment]
@mezz   ,OK with your exact setup in a new instance in MultiMC it still does same,  1188 to add 

```
[21:39:34] [main/INFO]: Loading tweak class name net.minecraftforge.fml.common.launcher.FMLTweaker
[21:39:34] [main/INFO]: Using primary tweak class name net.minecraftforge.fml.common.launcher.FMLTweaker
[21:39:34] [main/INFO]: Calling tweak class net.minecraftforge.fml.common.launcher.FMLTweaker
[21:39:34] [main/INFO]: Forge Mod Loader version 12.18.2.2114 for Minecraft 1.10.2 loading
[21:39:34] [main/INFO]: Java is Java HotSpot(TM) 64-Bit Server VM, version 1.8.0_101, running on Windows 10:amd64:10.0, installed at C:\Program Files\Java\jre1.8.0_101
[21:39:34] [main/INFO]: Loading tweak class name net.minecraftforge.fml.common.launcher.FMLInjectionAndSortingTweaker
[21:39:34] [main/INFO]: Loading tweak class name net.minecraftforge.fml.common.launcher.FMLDeobfTweaker
[21:39:34] [main/INFO]: Calling tweak class net.minecraftforge.fml.common.launcher.FMLInjectionAndSortingTweaker
[21:39:34] [main/INFO]: Calling tweak class net.minecraftforge.fml.common.launcher.FMLInjectionAndSortingTweaker
[21:39:34] [main/INFO]: Calling tweak class net.minecraftforge.fml.relauncher.CoreModManager$FMLPluginWrapper
[21:39:35] [main/INFO]: Found valid fingerprint for Minecraft Forge. Certificate fingerprint e3c3d50c7c986df74c645c0ac54639741c90a557
[21:39:35] [main/INFO]: Found valid fingerprint for Minecraft. Certificate fingerprint cd99959656f753dc28d863b46769f7f8fbaefcfc
[21:39:35] [main/INFO]: Calling tweak class net.minecraftforge.fml.relauncher.CoreModManager$FMLPluginWrapper
[21:39:35] [main/INFO]: Calling tweak class net.minecraftforge.fml.common.launcher.FMLDeobfTweaker
[21:39:35] [main/INFO]: Calling tweak class net.minecraftforge.fml.relauncher.CoreModManager$FMLPluginWrapper
[21:39:36] [main/INFO]: Loading tweak class name net.minecraftforge.fml.common.launcher.TerminalTweaker
[21:39:36] [main/INFO]: Calling tweak class net.minecraftforge.fml.common.launcher.TerminalTweaker
[21:39:36] [main/INFO]: Launching wrapped minecraft {net.minecraft.client.main.Main}
[21:39:36] [main/INFO]: Transforming net.minecraft.entity.player.EntityPlayer finished, added func_184613_cA() overriding EntityLivingBase
[21:39:36] [Client thread/INFO]: Setting user:
[21:39:37] [Client thread/INFO]: Transforming Class [net.minecraft.entity.ai.EntityAICreeperSwell], Method [func_75246_d]
[21:39:37] [Client thread/INFO]: Transforming net.minecraft.entity.ai.EntityAICreeperSwell Finished.
[21:39:37] [Client thread/INFO]: Transforming Class [net.minecraft.inventory.ContainerFurnace], Method [func_82846_b]
[21:39:37] [Client thread/INFO]: Transforming net.minecraft.inventory.ContainerFurnace Finished.
[21:39:37] [Client thread/WARN]: Skipping bad option: lastServer:
[21:39:38] [Client thread/INFO]: LWJGL Version: 2.9.1
[21:39:38] [Client thread/INFO]: [net.minecraftforge.fml.client.SplashProgress:start:219]: ---- Minecraft Crash Report ----

WARNING: coremods are present:
  EnderCorePlugin (EnderCore-1.10.2-0.4.1.56-beta.jar)
Contact their authors BEFORE contacting forge

// I feel sad now :(

Time: 10/27/16 9:39 PM
Description: Loading screen debug info

This is just a prompt for computer specs to be printed. THIS IS NOT A ERROR


A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- System Details --
Details:
    Minecraft Version: 1.10.2
    Operating System: Windows 10 (amd64) version 10.0
    Java Version: 1.8.0_101, Oracle Corporation
    Java VM Version: Java HotSpot(TM) 64-Bit Server VM (mixed mode), Oracle Corporation
    Memory: 1883795208 bytes (1796 MB) / 2392850432 bytes (2282 MB) up to 9320792064 bytes (8889 MB)
    JVM Flags: 3 total; -XX:HeapDumpPath=MojangTricksIntelDriversForPerformance_javaw.exe_minecraft.exe.heapdump -Xms2000m -Xmx8000m
    IntCache: cache: 0, tcache: 0, allocated: 0, tallocated: 0
    FML:
    Loaded coremods (and transformers):
EnderCorePlugin (EnderCore-1.10.2-0.4.1.56-beta.jar)
  com.enderio.core.common.transform.EnderCoreTransformer
    GL info: ' Vendor: 'NVIDIA Corporation' Version: '4.5.0 NVIDIA 373.06' Renderer: 'GeForce GTX 1080/PCIe/SSE2'
[21:39:38] [Client thread/INFO]: MinecraftForge v12.18.2.2114 Initialized
[21:39:38] [Client thread/INFO]: Replaced 231 ore recipes
[21:39:39] [Client thread/INFO]: Found 0 mods from the command line. Injecting into mod discoverer
[21:39:39] [Client thread/INFO]: Searching E:\Program Files (x86)\MultiMC\instances\1.10.2 TEST2\minecraft\mods for mods
[21:39:39] [Client thread/WARN]: ****************************************
[21:39:39] [Client thread/WARN]: * The modid JEI is not the same as it's lowercase version. Lowercasing will be enforced in 1.11
[21:39:39] [Client thread/WARN]: *  at net.minecraftforge.fml.common.FMLModContainer.sanityCheckModId(FMLModContainer.java:141)
[21:39:39] [Client thread/WARN]: *  at net.minecraftforge.fml.common.FMLModContainer.<init>(FMLModContainer.java:126)
[21:39:39] [Client thread/WARN]: *  at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
[21:39:39] [Client thread/WARN]: *  at sun.reflect.NativeConstructorAccessorImpl.newInstance(Unknown Source)
[21:39:39] [Client thread/WARN]: *  at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(Unknown Source)
[21:39:39] [Client thread/WARN]: *  at java.lang.reflect.Constructor.newInstance(Unknown Source)...
[21:39:39] [Client thread/WARN]: ****************************************
[21:39:39] [Client thread/INFO]: Forge Mod Loader has identified 6 mods to load
[21:39:39] [Client thread/INFO]: Attempting connection with missing mods [mcp, FML, Forge, actuallyadditions, endercore, JEI] at CLIENT
[21:39:39] [Client thread/INFO]: Attempting connection with missing mods [mcp, FML, Forge, actuallyadditions, endercore, JEI] at SERVER
[21:39:40] [Client thread/INFO]: Reloading ResourceManager: Default, FMLFileResourcePack:Forge Mod Loader, FMLFileResourcePack:Minecraft Forge, FMLFileResourcePack:Actually Additions, FMLFileResourcePack:EnderCore, FMLFileResourcePack:Just Enough Items
[21:39:40] [Client thread/INFO]: Processing ObjectHolder annotations
[21:39:40] [Client thread/INFO]: Found 423 ObjectHolder annotations
[21:39:40] [Client thread/INFO]: Identifying ItemStackHolder annotations
[21:39:40] [Client thread/INFO]: Found 0 ItemStackHolder annotations
[21:39:40] [Client thread/INFO]: Applying holder lookups
[21:39:40] [Client thread/INFO]: Holder lookups applied
[21:39:40] [Client thread/INFO]: Applying holder lookups
[21:39:40] [Client thread/INFO]: Holder lookups applied
[21:39:40] [Client thread/INFO]: Applying holder lookups
[21:39:40] [Client thread/INFO]: Holder lookups applied
[21:39:40] [Client thread/INFO]: Configured a dormant chunk cache size of 0
[21:39:40] [Forge Version Check/INFO]: [Forge] Starting version check at http://files.minecraftforge.net/maven/net/minecraftforge/forge/promotions_slim.json
[21:39:40] [Client thread/INFO]: Starting PreInitialization Phase...
[21:39:40] [Client thread/INFO]: Tesla not found! Skipping Tesla Power System integration.
[21:39:40] [Client thread/INFO]: Grabbing Configurations...
[21:39:40] [Client thread/INFO]: Initializing Tool Materials...
[21:39:40] [Client thread/INFO]: Initializing Armor Materials...
[21:39:40] [Client thread/INFO]: Initializing Blocks...
[21:39:40] [Forge Version Check/INFO]: [Forge] Found status: AHEAD Target: null
[21:39:40] [Client thread/INFO]: Initializing Items...
[21:39:40] [Client thread/INFO]: Initializing Fuelstuffs...
[21:39:40] [Client thread/INFO]: Initializing Update Checker...
[21:39:40] [Actually Additions Update Checker/INFO]: Starting Update Check...
[21:39:40] [Client thread/INFO]: PreInitializing ClientProxy...
[21:39:40] [Client thread/INFO]: PreInitialization Finished.
[21:39:40] [Client thread/INFO]: Applying holder lookups
[21:39:40] [Client thread/INFO]: Holder lookups applied
[21:39:40] [Client thread/INFO]: Injecting itemstacks
[21:39:40] [Client thread/INFO]: Itemstack injection complete
[21:39:40] [Actually Additions Update Checker/INFO]: Update Check done!
[21:39:40] [Actually Additions Update Checker/INFO]: Actually Additions is up to date!
[21:39:41] [Sound Library Loader/INFO]: Starting up SoundSystem...
[21:39:41] [Thread-9/INFO]: Initializing LWJGL OpenAL
[21:39:41] [Thread-9/INFO]: (The LWJGL binding of OpenAL.  For more information, see http://www.lwjgl.org)
[21:39:41] [Thread-9/INFO]: OpenAL initialized.
[21:39:41] [Sound Library Loader/INFO]: Sound engine started
[21:39:42] [Client thread/INFO]: Max texture size: 16384
[21:39:42] [Client thread/INFO]: Created: 16x16 textures-atlas
[21:39:42] [Client thread/INFO]: Transforming Class [net.minecraft.client.renderer.RenderItem], Method [func_180453_a]
[21:39:42] [Client thread/INFO]: Transforming net.minecraft.client.renderer.RenderItem Finished.
[21:39:42] [Client thread/INFO]: Starting Initialization Phase...
[21:39:42] [Client thread/INFO]: Initializing OreDictionary Entries...
[21:39:42] [Client thread/INFO]: Initializing Achievements...
[21:39:42] [Client thread/INFO]: Initializing GuiHandler...
[21:39:42] [Client thread/INFO]: Registering World Generator...
[21:39:42] [Client thread/INFO]: Registering TileEntities...
[21:39:42] [Client thread/INFO]: Initializing Crafting Recipes...
[21:39:42] [Client thread/INFO]: Initializing Dungeon Loot...
[21:39:42] [Client thread/INFO]: Initializing Entities...
[21:39:42] [Client thread/INFO]: Initializing ClientProxy...
[21:39:42] [Client thread/INFO]: Initialization Finished.
[21:39:42] [Client thread/INFO]: [Handlers] Registering handler SpecialTooltipHandler to busses: [FORGE, FML]
[21:39:42] [Client thread/INFO]: [Handlers] Registering handler XPBoostHandler to busses: [FORGE, FML]
[21:39:42] [Client thread/INFO]: [Handlers] Registering handler AutoSmeltHandler to busses: [FORGE, FML]
[21:39:42] [Client thread/INFO]: [Handlers] Registering handler EnchantTooltipHandler to busses: [FORGE, FML]
[21:39:43] [Client thread/INFO]: [Handlers] Registering handler Scheduler to busses: [FORGE, FML]
[21:39:43] [Client thread/INFO]: [Handlers] Registering handler FireworkHandler to busses: [FORGE, FML]
[21:39:43] [Client thread/INFO]: [Handlers] Registering handler JoinMessageHandler to busses: [FORGE, FML]
[21:39:43] [Client thread/INFO]: [Handlers] Registering handler RightClickCropHandler to busses: [FORGE, FML]
[21:39:43] [Client thread/INFO]: [Handlers] Registering handler OreDictTooltipHandler to busses: [FORGE, FML]
[21:39:43] [Client thread/INFO]: [Handlers] Registering handler ClientHandler to busses: [FORGE, FML]
[21:39:43] [Client thread/INFO]: Sending dummy event to all mods
[21:39:43] [Client thread/INFO]: Reloading ingame configs for modid: endercore
[21:39:43] [Client thread/INFO]: Injecting itemstacks
[21:39:43] [Client thread/INFO]: Itemstack injection complete
[21:39:43] [Client thread/INFO]: Starting PostInitialization Phase...
[21:39:43] [Client thread/INFO]: Initializing Village Addons...
[21:39:43] [Client thread/INFO]: Initializing Crusher Recipes...
[21:39:43] [Client thread/INFO]: Added 12 Crusher Recipes automatically: [oreGold -> 2xitem.actuallyadditions.itemDust@1, oreDiamond -> 2xitem.actuallyadditions.itemDust@2, oreEmerald -> 2xitem.actuallyadditions.itemDust@3, ingotIron -> 1xitem.actuallyadditions.itemDust@0, ingotGold -> 1xitem.actuallyadditions.itemDust@1, gemDiamond -> 1xitem.actuallyadditions.itemDust@2, gemEmerald -> 1xitem.actuallyadditions.itemDust@3, gemQuartz -> 1xitem.actuallyadditions.itemDust@5, gemPrismarine -> 1xitem.prismarineCrystals@0, gemLapis -> 1xitem.actuallyadditions.itemDust@4, oreQuartzBlack -> 2xitem.actuallyadditions.itemDust@7, gemQuartzBlack -> 1xitem.actuallyadditions.itemDust@7]
[21:39:43] [Client thread/WARN]: Couldn't add 0 Crusher Recipes automatically because the inputs were missing outputs: []
[21:39:43] [Client thread/INFO]: PostInitializing ClientProxy...
[21:39:43] [Actually Additions Special Fetcher/INFO]: Fetching Special People Stuff...
[21:39:43] [Client thread/INFO]: PostInitialization Finished.
[21:39:43] [Client thread/INFO]: Copying file endercore/config/cropConfig.json from jar
[21:39:43] [Client thread/INFO]: Starting JEI...
[21:39:43] [Actually Additions Special Fetcher/INFO]: Fetching Special People Stuff done!
[21:39:43] [Client thread/INFO]: Registering plugin: mezz.jei.plugins.vanilla.VanillaPlugin ...
[21:39:43] [Client thread/INFO]: Registered  plugin: mezz.jei.plugins.vanilla.VanillaPlugin in 60 ms
[21:39:43] [Client thread/INFO]: Registering plugin: de.ellpeck.actuallyadditions.mod.jei.JEIActuallyAdditionsPlugin ...
[21:39:43] [Client thread/INFO]: Registered  plugin: de.ellpeck.actuallyadditions.mod.jei.JEIActuallyAdditionsPlugin in 9 ms
[21:39:43] [Client thread/INFO]: Registering plugin: mezz.jei.plugins.jei.JEIInternalPlugin ...
[21:39:43] [Client thread/INFO]: Registered  plugin: mezz.jei.plugins.jei.JEIInternalPlugin in 1 ms
[21:39:43] [Client thread/INFO]: Building recipe registry...
[21:39:43] [Client thread/INFO]: Built    recipe registry in 75 ms
[21:39:43] [Client thread/INFO]: Building runtime...
[21:39:43] [Client thread/INFO]: Built    runtime in 31 ms
[21:39:43] [Client thread/INFO]: Building item filter...
[21:39:58] [Client thread/INFO]: Built    item filter in 15623 ms
[21:39:58] [Client thread/INFO]: Sending runtime to plugin: mezz.jei.plugins.vanilla.VanillaPlugin ...
[21:39:58] [Client thread/INFO]: Sending runtime to plugin: de.ellpeck.actuallyadditions.mod.jei.JEIActuallyAdditionsPlugin ...
[21:39:58] [Client thread/INFO]: Sending runtime to plugin: mezz.jei.plugins.jei.JEIInternalPlugin ...
[21:39:58] [Client thread/INFO]: Finished Starting JEI in 15853 ms
[21:39:59] [Client thread/INFO]: Forge Mod Loader has successfully loaded 6 mods
[21:39:59] [Client thread/INFO]: Reloading ResourceManager: Default, FMLFileResourcePack:Forge Mod Loader, FMLFileResourcePack:Minecraft Forge, FMLFileResourcePack:Actually Additions, FMLFileResourcePack:EnderCore, FMLFileResourcePack:Just Enough Items
[21:39:59] [Client thread/INFO]: SoundSystem shutting down...
[21:39:59] [Client thread/WARN]: Author: Paul Lamb, www.paulscode.com
[21:39:59] [Sound Library Loader/INFO]: Starting up SoundSystem...
[21:39:59] [Thread-12/INFO]: Initializing LWJGL OpenAL
[21:39:59] [Thread-12/INFO]: (The LWJGL binding of OpenAL.  For more information, see http://www.lwjgl.org)
[21:39:59] [Thread-12/INFO]: OpenAL initialized.
[21:40:00] [Client thread/INFO]: Max texture size: 16384
[21:40:00] [Sound Library Loader/INFO]: Sound engine started
[21:40:00] [Client thread/INFO]: Created: 1024x512 textures-atlas
[21:40:00] [Client thread/WARN]: Skipping bad option: lastServer:
[21:40:02] [Client thread/INFO]: Stopping!
[21:40:02] [Client thread/INFO]: SoundSystem shutting down...
[21:40:02] [Client thread/WARN]: Author: Paul Lamb, www.paulscode.com
```

**Comment by iTraumatik** [comment]
What launcher , java args , java mem settings are you using.  If your not using MultiMC i want to duplicate your setup.  My computer is i7 6700k @4.4 32gb 3466mhz  2 1080sli  installed on samsung 850pro ssd nothing else running just fyi

**Comment by mezz** [comment]
Using Curse launcher, I created a custom modpack and added those three mods. 8GB ram allocated to java.
I created a new MultiMC instance and had no issue either.
I even have a very similar but inferior computer setup, i7 2600K @3.4ghz, 32gb ram, GTX680 graphics, samsung 850evo ssd.

**Comment by iTraumatik** [comment]
@mezz TY very much.  Im spent for night.  Ill dig more tomorrow. Thanks again

**Comment by mezz** [comment]
This is starting to look like it's the same issue as with Optifine after all.

Looks like the slow-loading-with-Optifine issue has to do with the splash screen loading bars. If I disable the splash screen (`config/splash.properties`), everything loads quickly.

Try doing that, if it works then I'm pretty convinced that they are the same issue and you can take a long break from testing haha. Also it should be a decent workaround for now.

**Comment by iTraumatik** [comment]
WOW, that did the trick.  Thanks for finding this.  It got to the point it usually loads "ingredients" and in the log it looked like it loaded right thru instead of counting up. Gratz for this again.   Now i might be able to play and not worry about crashing and having to reload a loonnng time. 

OOPs....  My splash.properties is actually in minecraft/config/  not minecraft/config/forge..  ty

**Comment by mezz** [comment]
Ah fixed my comment to point to the splash.properties.
I'll mark this as a duplicate of #538, thanks for your patience.

**Comment by iTraumatik** [comment]
@mezz  Just a heads up, turning enabled=false in splash.properties is making mobs invisible.  2 creepers killed me twice so far, i could hear cows but not see them. I updated to forge 2116 and can see creepers, zombies, spiders now but got killed quickly.  If i remember correctly now , wasnt the invisible mobs and turning off this option a problem long ago with forge or something ? No Optifine installed
Also another heads up, using Curse launcher to launch MC was using about 2-3gb more java mem then just using MultiMC.  I checked task manager and happened every time.  I had to set java up to 12gb when using curse launcher to get to MC option screen but in MultiMC only need 8gb max.

EDIT ... sry jumped the gun, just loaded new world and can hear chickens cant see them. No optifine, i changed java settings in nvidia control panel, but when you quit and save then load back into the world i can see the chickens.  Another im at a loss.
